### PR TITLE
Updated workflow, fixed issue with db name

### PR DESCRIPTION
**Pull request by ieuna**

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -58,4 +62,28 @@ jobs:
       - name: "Validate Terrform format"
         working-directory: mxd
         run: |-
-          terraform fmt -check -recursive  
+          terraform fmt -check -recursive
+          
+  End-To-End-Tests:
```
Comment (hemantxpatel) [inline .github/workflows/verify.yaml:67]: Create a separate file for backend service. `verify-backend-service.yaml`

File: .github/workflows/verify.yaml [hunk 2]
```diff
@@ -58,4 +62,28 @@ jobs:
...
+    
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+          
+      - name: Change directory to backend-service and run tests
+        run: ./gradlew test
```
Comment (hemantxpatel) [inline .github/workflows/verify.yaml:88]: Run `./gradlew clean build test`, please revalidate the command from tractusx repo. Code should be built as well, not just test. And if we have some checkstyle, those should also be executed.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java [hunk 3]
```diff
@@ -78,10 +78,8 @@ public class BackendServiceExtension implements ServiceExtension {
     @Inject
     private TypeManager typeManager;
 
-    @Inject(required = false)
     private ContentStatementsService statements;
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:81]: This is an internal class, so it should not be injected. Just remove this field. Only external dependencies should be injected.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java [hunk 4]
```diff
@@ -78,10 +78,8 @@ public class BackendServiceExtension implements ServiceExtension {
...
 
-    @Inject(required = false)
     private TransferStatementsService transferStatementsService;
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:83]: Same comment as above.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java (hunk 3 above)
Comment (ieuna) [reply mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:81]: This injection has been removed already. It's red marked.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java (hunk 4 above)
Comment (ieuna) [reply mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:83]: This injection has been removed already. It's red marked.

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (ieuna) [reply .github/workflows/verify.yaml:67]: but the reviewer specified- "in the verify.yaml workflow"

File: .github/workflows/verify.yaml (hunk 2 above)
Comment (ieuna) [reply .github/workflows/verify.yaml:88]: this is what I followed: [Connector Workflow](https://github.com/eclipse-edc/Connector/blob/4f3621811a8143167554206e1d9dbae2d60c9d23/.github/workflows/verify.yaml#L74)

this is how the connector configures E2E workflow

File: .github/workflows/verify-backend-service.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: password
+          POSTGRES_DB: backendservicedb
+        ports:
+          - 5432:5432
```
Comment (hemantxpatel) [inline .github/workflows/verify-backend-service.yaml:47]: We run an in memory postgres instance while running tests, right, do we still need it?

File: .github/workflows/verify-backend-service.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+          
+      - name: Change directory to backend-service and run tests
```
Comment (hemantxpatel) [inline .github/workflows/verify-backend-service.yaml:56]: Name should be `Run E2E tests`.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java [hunk 7]
```diff
@@ -200,14 +194,11 @@ private String getDataSourceName(ServiceExtensionContext context) {
     }
 
     private ContentStatementsService getStatementImpl() {
-        return statements != null ? statements : new ContentStatementsServiceImpl() {
-        };
+        return new ContentStatementsServiceImpl() {};
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:197]: `{}` should not be needed at the end of the line.

File: mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java [hunk 8]
```diff
@@ -200,14 +194,11 @@ private String getDataSourceName(ServiceExtensionContext context) {
...
     }
 
     private TransferStatementsService getTransferStatementImpl() {
-        return transferStatementsService != null ? transferStatementsService :
-                new TransferStatementsServiceImpl() {
-                };
+        return new TransferStatementsServiceImpl() {};
```
Comment (hemantxpatel) [inline mxd/backend-service/src/main/java/org/eclipse/tractusx/mxd/backendservice/BackendServiceExtension.java:201]: Same as above

**Review by hemantxpatel — APPROVED** [review]

**Pull request merged**